In [4]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("Supermart Grocery Sales - Retail Analytics Dataset.csv")

print(df.head())
print(df.info())


  Order ID Customer Name          Category      Sub Category         City  \
0      OD1        Harish      Oil & Masala           Masalas      Vellore   
1      OD2         Sudha         Beverages     Health Drinks  Krishnagiri   
2      OD3       Hussain       Food Grains      Atta & Flour   Perambalur   
3      OD4       Jackson  Fruits & Veggies  Fresh Vegetables   Dharmapuri   
4      OD5       Ridhesh       Food Grains   Organic Staples         Ooty   

   Order Date Region  Sales  Discount  Profit       State  
0  11-08-2017  North   1254      0.12  401.28  Tamil Nadu  
1  11-08-2017  South    749      0.18  149.80  Tamil Nadu  
2  06-12-2017   West   2360      0.21  165.20  Tamil Nadu  
3  10-11-2016  South    896      0.25   89.60  Tamil Nadu  
4  10-11-2016  South   2355      0.26  918.45  Tamil Nadu  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         ------

In [5]:

# تحويل الأعمدة الرقمية إلى أرقام
df["Sales"] = pd.to_numeric(df["Sales"], errors="coerce")
df["Profit"] = pd.to_numeric(df["Profit"], errors="coerce")
df["Discount"] = pd.to_numeric(df["Discount"], errors="coerce")


In [7]:

# حساب إجمالي المبيعات
total_sales = df["Sales"].sum()
total_sales

np.int64(14956982)

In [8]:
# الوصف الإحصائي لقيمة المبيعات
sales_description = df["Sales"].describe()
sales_description

,Sales
count,9994.000000
mean,1496.596158
std,577.559036
min,500.000000
25%,1000.000000
50%,1498.000000
75%,1994.750000
max,2500.000000


In [9]:
# حساب متوسط قيمة الطلب
average_sales = df["Sales"].mean()
average_sales

np.float64(1496.5961576946167)

In [10]:
# حساب الوسيط
median_sales = df["Sales"].median()
median_sales

1498.0

In [11]:

# حساب أقل وأعلى قيمة
minimum_sales = df["Sales"].min()
maximum_sales = df["Sales"].max()
print("أقل قيمة طلب:", minimum_sales)
print("أعلى قيمة طلب:", maximum_sales)

أقل قيمة طلب: 500
أعلى قيمة طلب: 2500


In [12]:
# حساب إجمالي الأرباح
total_profit = df["Profit"].sum()
print("إجمالي الأرباح:", total_profit)

إجمالي الأرباح: 3747121.1999999997


ظهر تحليل بيانات المبيعات أن إجمالي المبيعات بلغ 14 956,982 من خلال 9,994 طلبًا وبلغ متوسط قيمة الطلب 1,496.60 بينما بلغ الوسيط 1,498 مما يشير إلى أن قيم الطلبات تتمركز حول هذا المستوى تراوحت قيم المبيعات بين 500 و2,500 مع انحراف معياري قدره 577.56 مما يدل على وجود اختلاف ملحوظ في قيمة الطلبات

كما بلغ إجمالي الأرباح 3,747,121.20. توضح البيانات وجود حجم مبيعات جيد ويمكن استخدام هذا التحليل لفهم أداء المتجر ومقارنة المبيعات بين الفئات المختلفة واتخاذ قرارات أفضل بشأن المنتجات والمبيعات.